src/Conversion/ONNXToTOSA/Math/Gemm.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Gemm.cpp - Gemm Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX Gemm operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "llvm/ADT/SmallVector.h"
+#include <cstdint>
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+
+
+class ONNXGemmOpLoweringToTOSA : public OpConversionPattern<ONNXGemmOp> {
+public:
+  using OpConversionPattern<ONNXGemmOp>::OpConversionPattern;
+  LogicalResult rewriteToTosaFC(ONNXGemmOp op, OpAdaptor adaptor,
```

Comment:
I wonder if splitting this in `check()`and `rewrite()`is helpful, or if a `matchAndRewriteToFC()` would work better.